### DISCUSSION TITLE
Questions concerning this project

### DISCUSSION THREAD

**Discussion by maxmontgmx** [ni/grpc-teststand-api#18] [General]
Hi,
I plan to design a TestStand operator interface for our manufacturing with .NET framework (C# language)
I started with UI Sample provided with TestStand setup.
And yesterday, I discovered this new project.
My questions are : 
- will this feature be supported and official in the futur?
- if there is no need of multi client, does it make sense to use the remoting API instead of traditional API?
- do all traditional API functions be avialable with remoting API? or does it exist some limitations?
It's an open discussion to help me choosing the best approch for a new development and not making a mistake.
Thanks in advance for your reponses/comments

**Comment by joseahdz** [comment]
Hi maxmontgmx,
These answers might be a little late, but these are my answers to your questions.  

- We don't have a date for an official release of the TestStand gRPC API yet.  
- If you plan to always run the operator interface in the same machine where the tests are running, there is no need to us the gRPC API.  If you are planning on creating a TestStand Windows service or a headless server that will run your tests, I highly recommend using the gRPC API to allow remote clients to start tests and monitor them.  You can also write a web application that communicates with the headless server using the gRPC API.  
- The TestStand gRPC API provides all the functionality that the traditional (interop) APIs provide.  I highly recommend reading the Docs\APIMapping.md file.  It gives you an overview of how we map the interop APIs to the gRPC APIs.
